src/util.cpp

```diff
@@ -227,33 +244,71 @@ void OpenDebugLog()
     vMsgsBeforeOpenLog = NULL;
 }
 
-bool LogAcceptCategory(const char* category)
+/*
+ * Update the 'enabled' flags of all (reached, known) log-fast entries
+ * according to current sets of enabled categories and lines. This can
+ * be somewhat expensive but is only called when the set of enabled
+ * categories or lines changes (by the api). This is never called in-
+ * context of a thread calling LogPrint() or LogAcceptCategory().
+ */
+static bool logFastEnabled(const LogFastEntry *lfe)
 {
-    if (category != NULL)
-    {
-        if (!fDebug)
-            return false;
-
-        // Give each thread quick access to -debug settings.
```

Comment:
This is a hack, and was eliminated in the same commit that replaced string categories with bitmap categories. It would be nice for us to get rid of too.